lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php

```diff
@@ -53,14 +34,14 @@ public function connect(array $params, $username = null, $password = null, array
     /**
      * Constructs the MySql PDO DSN.
      *
-     * @param array $params
+     * @param mixed[] $params
      *
      * @return string The DSN.
      */
     protected function constructPdoDsn(array $params)
     {
         $dsn = 'mysql:';
-        if (isset($params['host']) && $params['host'] != '') {
+        if (isset($params['host']) && $params['host'] !== '') {
```

Comment:
This can be considered a BC break. Could be replaced with `... && strcmp($params['host'], '') === 0`.